fintech-examples/fintech-ui/src/app/bank/list-transactions/list-transactions.component.ts

```diff
@@ -18,23 +19,46 @@ export class ListTransactionsComponent implements OnInit, OnDestroy {
   transactionsLists: Array<Array<TransactionDetails>>;
   transactionsListNames = ['Pending', 'Booked'];
 
-  constructor(private route: ActivatedRoute, private aisService: AisService) {}
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private aisService: AisService) {
+  }
 
   ngOnInit() {
-    this.bankId = this.route.parent.parent.parent.snapshot.paramMap.get('bankid');
-    console.log('list-transactions for bankid', this.bankId);
-    this.accountId = this.route.snapshot.paramMap.get('accountid');
-    console.log('list-transactions for accountid', this.accountId);
+    this.route.parent.parent.parent.paramMap.subscribe(p => {
+        this.bankId = p.get('bankid');
+        console.log('list-transactions for bankid', this.bankId);
+      }
+    );
+    this.route.paramMap.subscribe(p => {
+        this.accountId = p.get('accountid');
+        console.log('list-transactions for accountid', this.accountId);
+      }
+    );
 
     this.transactionsLists = [];
-    console.log('ask for transactions for ', this.accountId);
-    this.accountsSubscription = this.aisService.getTransactions(this.bankId, this.accountId).subscribe(transactions => {
-      this.transactionsLists = [transactions.pending, transactions.booked];
-      this.makeVisible = true;
-    });
+    console.log('ON INIT LTX ask for transactions for ', this.accountId);
```

Comment:
We will probably have a bug here because the instructions above are async. Think about what will happen if line 31 is executed before either `line 29` or `line 34` finished

Reply:
yep, you are right. fixed.